dd-java-agent/appsec/src/main/java/com/datadog/appsec/config/LegacyConfigSerializer.java

```
@@ -0,0 +1,159 @@
+          for (Object obj : step.transformers) {
+            gen.writeObject(obj);
+          }
+          gen.writeEndArray();
+        }
+
+        gen.writeEndObject();
+      }
+      gen.writeEndArray();
+      gen.writeEndObject();
+    }
+    gen.writeEndArray();
+
+    // Manifest
+    gen.writeObjectFieldStart("manifest");
+    for (String parameter : parameters) {
+      gen.writeObjectFieldStart(parameter);
+      gen.writeStringField("inherit_from", parameter);
+      gen.writeBooleanField("run_on_value", true);
+      gen.writeBooleanField("run_on_key", false);
+      gen.writeEndObject();
+    }
+    gen.writeEndObject();
+
+    gen.writeArrayFieldStart("flows");
+
+    gen.writeStartObject();
+    gen.writeStringField("name", "blocking");
+    gen.writeArrayFieldStart("steps");
+
+    // Block step
+    if (!blockingRulesId.isEmpty()) {
+      gen.writeStartObject();
+      gen.writeStringField("id", "start");
+      gen.writeArrayFieldStart("rule_ids");
+      for (String s : blockingRulesId) {
+        gen.writeObject(s);
+      }
+      gen.writeEndArray();
+      gen.writeStringField("on_match", "exit_block");
+      gen.writeEndObject();
+    }
+
+    // Monitor step
+    if (!passRulesId.isEmpty()) {
+      gen.writeStartObject();
+      gen.writeStringField("id", "start");
+      gen.writeArrayFieldStart("rule_ids");
+      for (String s : passRulesId) {
+        gen.writeObject(s);
+      }
+      gen.writeEndArray();
+      gen.writeStringField("on_match", "exit_monitor");
+      gen.writeEndObject();
+    }
+
+    gen.writeEndArray();
+    gen.writeEndObject();
+
+    gen.writeEndArray();
+    gen.writeEndObject();
+  }
+
+  private String md5(String s) {
+    try {
+      MessageDigest md = MessageDigest.getInstance("MD5");
+      md.update(s.getBytes(), 0, s.length());
+      return new BigInteger(1, md.digest()).toString(16);
+    } catch (NoSuchAlgorithmException e) {
+      e.printStackTrace();
```

Log instead. Also it seems this should never happen anyway, so can just wrap it in a some kind of runtime exception and let it propagate, esp. because now you can return null and the caller doesn't handle nulls correctly.